io_scene_godot/converters/multimesh.py

```diff
@@ -0,0 +1,185 @@
+import math
+import bpy
+import mathutils
+
+from ..structures import (
+    Array, NodeTemplate, InternalResource, fix_matrix)
+from .mesh import ArrayMeshResourceExporter
+
+
+def export_multimesh_node(escn_file, export_settings,
+                          obj, parent_gd_node):
+    """Export blender particle to a MultiMeshInstance"""
+    context = bpy.context
+    dg = context.evaluated_depsgraph_get()
+    ob = context.object.evaluated_get(dg)
+
+    ps = ob.particle_systems.active
+    if (ps.settings.instance_collection and
+            ps.settings.instance_collection.all_objects[0]):
+        instance_object = ps.settings.instance_collection.all_objects[0]
+    elif ps.settings.instance_object:
+        instance_object = ps.settings.instance_object
+
+    multimeshnode = NodeTemplate(ps.name, 'MultiMeshInstance', parent_gd_node)
+
+    # Export instance mesh resource first
+    instance_mesh_exporter = ArrayMeshResourceExporter(instance_object)
+
+    mesh_id = instance_mesh_exporter.export_mesh(escn_file, export_settings)
+
+    multimeshExporter = MultiMeshResourceExporter(obj, mesh_id, ps)
+
+    multimeshid = multimeshExporter.export_multimesh(
+        escn_file, export_settings, ps.name)
+
+    multimeshnode['multimesh'] = 'SubResource({})'.format(multimeshid)
+    multimeshnode['visible'] = obj.visible_get()
+
+    escn_file.add_node(multimeshnode)
+    return multimeshnode
+
+
+def has_particle(node):
+    """Returns True if the object has particles"""
+    context = bpy.context
+    dg = context.evaluated_depsgraph_get()
+    ob = context.object.evaluated_get(dg)
+
+    return len(ob.particle_systems) > 0
+
+
+class MultiMeshResourceExporter:
+    """Export a multimesh resource from a blender mesh object"""
+
+    def __init__(self, mesh_object, instance_mesh_id, particle_system):
+        # blender multimesh object
+        self.object = mesh_object
+        self.instance_mesh_id = instance_mesh_id
+        self.particle_system = particle_system
+
+        self.mesh_resource = None
+
+    def export_multimesh(self, escn_file, export_settings, particle_name):
+        """Saves a mesh into the escn file"""
+        converter = MultiMeshConverter(self.particle_system)
+        key = MultiMeshResourceKey(

```

Comment:
same as above,Because if "Render-Render As" are switched to "Collection",there maybe several objects instanced in one particle system.so when we choose one of particle system,maybe different object should be instance,I admit just use hash from Bpy is good for now. I thought too much maybe give an option that can select which object in "collection" to export in future haha.

Reply:
Very good point, if in the future we have custom logic for each object referring an instanced particle system to export to a MutltiMeshInstance, then a bpy ParticleSystem hash might not be enough.

I am okay with current implementation and I think it works well. However, I still feel that the `MultiMeshResourceKey` is a little bit overcomplicated. In my opinion, make the key refer to bpy ParticleSystem object instead of the particle system name is better. The later is guaranteed unique. The `gd_rsc_type` seems unnecessary, only `MultiMesh` is involved here.

Anyway I will merge it whether you change it or not.

Reply:
Thanks and not change for now!